Hazel/src/Hazel/Core/Log.h

```diff
@@ -33,4 +33,4 @@ namespace Hazel {
 #define HZ_INFO(...)          ::Hazel::Log::GetClientLogger()->info(__VA_ARGS__)
 #define HZ_WARN(...)          ::Hazel::Log::GetClientLogger()->warn(__VA_ARGS__)
 #define HZ_ERROR(...)         ::Hazel::Log::GetClientLogger()->error(__VA_ARGS__)
-#define HZ_CRITICAL(...)      ::Hazel::Log::GetClientLogger()->critical(__VA_ARGS__)
\ No newline at end of file
+#define HZ_CRITICAL(...)      ::Hazel::Log::GetClientLogger()->critical(__VA_ARGS__); throw std::runtime_error("Fatal error.");
```

Comment:
You should avoid adding multiple expressions to a macro, this would break

if (something) HZ_CRITICAL(....)

Reply:
You better should said that nobody should use IF expression without brackets, because it can bring a lot of unexpected bugs in your code.

I suggest rewrite this macro to:
```suggestion
#define HZ_CRITICAL(...)     { ::Hazel::Log::GetClientLogger()->critical(__VA_ARGS__); throw std::runtime_error("Fatal error."); }
```

Reply:
'nobody' might be easy when you control what is written but harder with more users, the alternative suggested solution now also means this is not mean to have a trailing semicolon unlike the other macros.

Also this introduces exceptions which the code base does not already use and is not designed to be exception safe.

Reply:
About the exception: agreed.
About the the if without {} being prohibited: that's not the only situation where macros like this one could break things... of course, this is a "final/error" macro, but if you have a macro that actually do something (calls another function, declares a var, or whatever) you could make use of it inside a while(<macro>). In order to make it fine, the solution would be:
```
({ ::Hazel::Log::GetClientLogger()->critical(__VA_ARGS__); throw std::runtime_error("Fatal error."); })
```
or
```
do { ::Hazel::Log::GetClientLogger()->critical(__VA_ARGS__); throw std::runtime_error("Fatal error."); } while(0)
```
now you must add a trailing semicolon and you have it guarded for both if() without brackets and inline conditions situations.